core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTabletMap.java

```diff
@@ -0,0 +1,267 @@
+package com.datastax.oss.driver.internal.core.metadata;
+
+import com.datastax.oss.driver.api.core.data.TupleValue;
+import com.datastax.oss.driver.api.core.metadata.HostShardPair;
+import com.datastax.oss.driver.api.core.metadata.KeyspaceTableNamePair;
+import com.datastax.oss.driver.api.core.metadata.Tablet;
+import com.datastax.oss.driver.api.core.metadata.TabletMap;
+import com.datastax.oss.driver.shaded.guava.common.annotations.Beta;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Holds currently known tablet mappings. Updated lazily through received custom payloads described
+ * in Scylla's CQL protocol extensions (tablets-routing-v1).
+ */
+@Beta
+public class DefaultTabletMap implements TabletMap {
+  private static final Logger logger = LoggerFactory.getLogger(DefaultTabletMap.class);
+
+  private final ConcurrentMap<KeyspaceTableNamePair, NavigableSet<Tablet>> mapping;
+
+  public DefaultTabletMap(ConcurrentMap<KeyspaceTableNamePair, NavigableSet<Tablet>> mapping) {
+    this.mapping = mapping;
+  }
+
+  public static DefaultTabletMap emptyMap() {
+    return new DefaultTabletMap(new ConcurrentHashMap<>());
+  }
+
+  @Override
+  public Map<KeyspaceTableNamePair, NavigableSet<Tablet>> getMapping() {
+    return mapping;
+  }
+
+  @Override
+  public Set<UUID> getReplicas(String keyspace, String table, long token) {
+    KeyspaceTableNamePair key = new KeyspaceTableNamePair(keyspace, table);
+
+    if (mapping == null) {
+      logger.trace("This tablets map is null. Returning empty set.");
+      return Collections.emptySet();
+    }
```

Comment:
Why it needs to be checked here? Can't you make sure that this field is never null?

Reply:
Yeah, seems unnecessary. Removed this.